assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -712,3 +713,300 @@ export function provideAnalytics4MockReport( registry, options ) {
 		.dispatch( MODULES_ANALYTICS_4 )
 		.receiveGetReport( getAnalytics4MockResponse( options ), { options } );
 }
+
+/**
+ * Creates all combinations of pivot dimension values that we expect in a pivot report response.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<string>} dimensionValues An array of valid dimension names.
+ * @return {Array.<Array>} An array of all possible combinations of dimension values.
+ */
+function createPivotDimensionCombinations( dimensionValues ) {
+	// Find every combination of each array of dimensionValues.
+	return dimensionValues.reduce(
+		( acc, dimensionValue ) =>
+			acc.flatMap( ( combination ) =>
+				dimensionValue.map( ( value ) => [ ...combination, value ] )
+			),
+		[ [] ]
+	);
+}
+
+/**
+ * Sorts pivot report rows and returns them.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>} rows    Array of rows to sort.
+ * @param {Array.<Object>} metrics Array of report metrics.
+ * @param {Array.<Object>} pivots  Report pivots which may include orderby values.
+ * @return {Array.<Object>} Sorted rows.
+ */
+export function sortPivotRows( rows, metrics, pivots ) {
+	// Extract all dimensions from the pivots orderby values if present.
+	const orderby = pivots
+		.filter( ( pivot ) => pivot.orderby )
+		.reduce( ( acc, pivot ) => {
+			pivot.orderby.forEach( ( orderByItem ) => {
+				acc.push( orderByItem );
+			} );
+			return acc;
+		}, [] );
+
+	if ( orderby.length === 0 ) {
+		return rows;
+	}
+
+	// For each pivot orderby, sort the rows by the given sorting metric and desc value.
+	return orderby.reduce( ( acc, orderbyItem ) => {
+		// Our mock metrics are returned in the rows in the order the metrics are given
+		// in the report config so we can take the index in the report array to find the
+		// column to sort by.
+		const metricIndex = metrics.findIndex(
+			( metric ) => metric.name === orderbyItem.metric.metricName
+		);
+
+		return acc.sort( ( rowA, rowB ) =>
+			orderbyItem.desc
+				? rowB.metricValues[ metricIndex ].value -
+				  rowA.metricValues[ metricIndex ].value
+				: rowA.metricValues[ metricIndex ].value -
+				  rowB.metricValues[ metricIndex ].value
+		);
```

Comment:
This is tough to read; it's technically all one statement with an implicit return and multiline-ternary. It winds up being tough to glance at and figure out. How about:

```suggestion
		return acc.sort( ( rowA, rowB ) => {
			if ( orderbyItem.desc ) {
				return rowB.metricValues[ metricIndex ].value - rowA.metricValues[ metricIndex ].value;
			}

			return rowA.metricValues[ metricIndex ].value - rowB.metricValues[ metricIndex ].value;
		} );
```

(Arguably you could also just sort it one way, then reverse the array if `orderbyItem.desc` is set, but technically I guess this is more efficient and it's not any harder to read 😄 )